plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -135,7 +137,13 @@ public TableMetadata refresh()
     public TableMetadata refresh(boolean invalidateCaches)
     {
         if (location.isPresent()) {
-            refreshFromMetadataLocation(null);
+            try {
+                String newLocation = fixBrokenMetadataLocation(getRefreshedLocation(invalidateCaches));
+                refreshFromMetadataLocation(newLocation);
+            }
+            catch (TableNotFoundException | UnknownTableTypeException e) {
+                refreshFromMetadataLocation(null);
```

Comment:
Could you leave a code comment why we need `refreshFromMetadataLocation` call for these exceptions?

Reply:
`refreshFromMetadataLocation(null)` is a noop for null location, but it sets `shouldRefresh=false`